akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -107,7 +106,8 @@ private[http] object HttpServerBluePrint {
 
     val rendererPipeline =
       Flow[ResponseRenderingContext]
-        .via(Flow[ResponseRenderingContext].transform(() ⇒ new ErrorsTo500ResponseRecovery(log)).named("recover")) // FIXME: simplify after #16394 is closed
+        .via(Flow[ResponseRenderingContext].transform(() ⇒ new ErrorResponseRecovery(log,
+          settings)).named("recover")) // FIXME: simplify after #16394 is closed
```

Comment:
Ah, yes. Good catch.
I'll apply the simplification then as well, in a separate commit.